landing-zone/variables.tf

```diff
@@ -0,0 +1,1529 @@
+          algorithm         = string
+          protocol          = string
+          health_delay      = number
+          health_retries    = number
+          health_timeout    = number
+          health_type       = string
+          pool_member_port  = string
+          security_group = optional(
+            object({
+              name = string
+              rules = list(
+                object({
+                  name      = string
+                  direction = string
+                  source    = string
+                  tcp = optional(
+                    object({
+                      port_max = number
+                      port_min = number
+                    })
+                  )
+                  udp = optional(
+                    object({
+                      port_max = number
+                      port_min = number
+                    })
+                  )
+                  icmp = optional(
+                    object({
+                      type = number
+                      code = number
+                    })
+                  )
+                })
+              )
+            })
+          )
+        })
+      ))
+    })
+  )
+  default = []
+
+  validation {
+    error_message = "Image names for F5 VSI must be one of [`f5-bigip-15-1-5-1-0-0-14-all-1slot`,`f5-bigip-15-1-5-1-0-0-14-ltm-1slot`, `f5-bigip-16-1-2-2-0-0-28-ltm-1slot`,`f5-bigip-16-1-2-2-0-0-28-all-1slot`]."
+    condition = length(
+      [
+        for f5_vsi in var.f5_vsi :
+        f5_vsi if !contains(
+          [
+            "f5-bigip-15-1-5-1-0-0-14-all-1slot",
+            "f5-bigip-15-1-5-1-0-0-14-ltm-1slot",
+            "f5-bigip-16-1-2-2-0-0-28-ltm-1slot",
+            "f5-bigip-16-1-2-2-0-0-28-all-1slot"
+          ],
+          f5_vsi.f5_image_name
+        )
+      ]
+    ) == 0
+  }
+}
+
+variable "f5_template_data" {
+  description = "Data for all f5 templates"
+  sensitive   = true
+  type = object({
+    tmos_admin_password     = optional(string)
+    license_type            = optional(string)
+    byol_license_basekey    = optional(string)
+    license_host            = optional(string)
+    license_username        = optional(string)
+    license_password        = optional(string)
+    license_pool            = optional(string)
+    license_sku_keyword_1   = optional(string)
+    license_sku_keyword_2   = optional(string)
+    license_unit_of_measure = optional(string)
+    do_declaration_url      = optional(string)
+    as3_declaration_url     = optional(string)
+    ts_declaration_url      = optional(string)
+    phone_home_url          = optional(string)
+    template_source         = optional(string)
+    template_version        = optional(string)
+    app_id                  = optional(string)
+    tgactive_url            = optional(string)
+    tgstandby_url           = optional(string)
+    tgrefresh_url           = optional(string)
+  })
+  default = {
+    tmos_admin_password = "Iamapassword2ru"
```

Comment:
For discussion, do we want to hardcode a password value here?

Reply:
I don't think we should have a default password here. If the user forgets to change it then they will be left vulnerable.

Reply:
you can take out the password